packages/polaris-viz/src/components/LineChartRelational/components/CustomLegend/CustomLegend.tsx

```diff
@@ -17,7 +32,27 @@ export function CustomLegend({
   getColorVisionStyles,
   seriesNameFormatter,
   theme,
+  activeIndex,
+  legendItemDimensions,
+  renderHiddenLegendLabel = (count) => `+${count} more`,
 }: Props) {
+  const {containerBounds} = useChartContext();
+  const deduplicatedData = deduplicateByRelatedIndex(data);
```

Comment:
We need to remove duplicatedData that doesn't appear on the legends, e.g. the percentiles. We have two entries in the data for the percentilce, but we need just one to perform calculations to hide/show the legends on smaller screens.